### PR TITLE
rocksdb: use features to control linking

### PR DESCRIPTION
**Pull request by BusyJay**
ref ngaut/rust-rocksdb#52

@zhangjinpeng1987 @disksing PTAL

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -1,5 +1,13 @@
 ENABLE_FEATURES ?= default
 
+ifeq ($(ROCKSDB_SYS_STATIC),1)
```
Comment (siddontang) [inline Makefile]: Now we have a `static_release` tag, so I think it is better to handle these in `static_release` directly.

For outside, `make release` is to build dynamic binary, and `make static_release` is to build static binary, no need to pass environment explicitly.

Comment (BusyJay) [inline Makefile]: For backward compatibility.

Comment (siddontang) [inline Makefile]: Got it

**Comment by siddontang** [comment]
LGTM

**Comment by zhangjinpeng1987** [comment]
LGTM

**Pull request merged**